src/main/java/org/example/coffeeshop/service/impl/UserServiceImpl.java

```diff
@@ -15,15 +21,51 @@ public class UserServiceImpl implements UserService {
 
     private final UserRepository userRepository;
     private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
 
     @Override
-    public User save(User user) {
-        user.setPassword(passwordEncoder.encode(user.getPassword()));
-        return userRepository.save(user);
+    public Optional<User> findByEmail(String email) {
+        return userRepository.findByEmail(email);
     }
 
     @Override
-    public Optional<User> findByEmail(String email) {
-        return userRepository.findByEmail(email);
+    public String createEmployee(RegisterUserRequestDto registerUserRequestDto, BindingResult result,
+                                 RedirectAttributes redirectAttributes, Model model) {
+        if (result.hasErrors()) {
+            model.addAttribute("user", registerUserRequestDto);
+            return "register";
+        }
+
+        if (!registerUserRequestDto.getPassword().equals(registerUserRequestDto.getConfirmPassword())) {
+            result.rejectValue("password", "error.userRequestDto", "Passwords do not match.");
+            model.addAttribute("user", registerUserRequestDto);
+            return "register";
+        }
+
+        if (registerUserRequestDto.getEmail() == null || !registerUserRequestDto.getEmail().matches("^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.(com|ru)$")) {
+            result.rejectValue("email", "error.userRequestDto", "The email must be in a valid format and end with '.com' or '.ru'.");
+            model.addAttribute("user", registerUserRequestDto);
+            return "register";
+        }
+
+        if (!registerUserRequestDto.getPassword().matches("^[A-Z].{7,}$")) {
+            result.rejectValue("password", "error.userRequestDto", "The password must start with an uppercase letter and be at least 8 characters long.");
+            model.addAttribute("user", registerUserRequestDto);
+            return "register";
+        }
```

Comment:
Validation should happen with Spring Validation, not by custom implementation
https://www.baeldung.com/spring-boot-bean-validation